client/src/components/StyledButton.tsx

```diff
@@ -73,20 +80,28 @@ const StyledButton = React.forwardRef<HTMLDivElement, StyledButtonProps>(({
         height: StyledButtonHeight,
         pt: hover ? 0 : shift,
         pb: shift,
-        cursor: 'pointer',
+        cursor: isButtonDisabled ? 'not-allowed' : 'pointer',
         backgroundImage: `url('${swapImage ? hover ? themedImageHover : themedImage : themedImage}')`,
         backgroundRepeat: 'no-repeat',
         backgroundSize: '100%',
         filter: `${shadow ? `drop-shadow(4px 4px 0px ${shadowColor})` : ''} ${contrast && hover ? 'contrast(90%)' : ''}`,
         fontVariant: 'small-caps',
         fontWeight: 'bold',
         color: 'secondary.main',
+        position: 'relative',
         ...sx,
         mb: (hover && shiftMargin) ? 1 : undefined,
       }}
       {...rest}
     >
-      {children}
+      {loading && (
+        <Box sx={{ position: 'absolute', left: '50%', top: '50%', transform: 'translate(-50%, -50%)' }}>
+          <Loader size={16} color="inherit" />
+        </Box>
+      )}
+      <Box sx={{ visibility: loading ? 'hidden' : 'initial', }}>
+        {children}
+      </Box>
```

Comment:
Same notes as the FantasyButton, only display the loader, not the children if it's loading